src/install.ts

```diff
@@ -1,24 +1,31 @@
 import execa from 'execa'
-import { detectPackageManager } from '.'
+import { detectPackageManager, isYarnBerry } from '.'
 
 export interface InstallPackageOptions {
   cwd?: string
   dev?: boolean
   silent?: boolean
   packageManager?: string
+  packageManagerVersion?: string
   preferOffline?: boolean
   additionalArgs?: string[]
 }
 
 export async function installPackage(names: string | string[], options: InstallPackageOptions = {}) {
   const agent = options.packageManager || await detectPackageManager(options.cwd) || 'npm'
+
   if (!Array.isArray(names))
     names = [names]
 
   const args = options.additionalArgs || []
 
-  if (options.preferOffline)
-    args.unshift('--prefer-offline')
+  if (options.preferOffline) {
+    // yarn v2+ uses --cached option instead of --prefer-offline
+    if (agent === 'yarn' && await isYarnBerry())
```

Comment:
Should we also check `packageManagerVersion`?

Reply:
It is not necessary. In yarn v1 there is no `.yarn` folder.

If we have a packageManager entry for yarn v2+ in package.json file without a `.yarn` folder it won't work because all yarn files are inside it.
I just checked what happens if I remove the entry from package.json: yarn still detects the v2+ version by its `.yarn` folder.
So the `.yarn` folder detection should be sufficient to detect if we still have yarn v1 or yarn v2+.

Reply:
I just double checked the yarn documentation for v1 and it seems that there is a way to create a .yarn folder with `yarn policies` in v1: https://classic.yarnpkg.com/en/docs/cli/policies/

So I will adjust the code to also check for an "packageManager" entry in package.json.